app/src/main/java/com/teamtwo/apilol/sensor/GpsLocation.kt

```diff
@@ -0,0 +1,72 @@
+package com.teamtwo.apilol.sensor
+
+import android.app.Activity
+import android.content.Context
+import android.content.Context.LOCATION_SERVICE
+import android.location.Location
+import android.location.LocationListener
+import android.location.LocationManager
+import android.os.Bundle
+import com.karumi.dexter.Dexter
+import com.karumi.dexter.PermissionToken
+import com.karumi.dexter.listener.PermissionDeniedResponse
+import com.karumi.dexter.listener.PermissionGrantedResponse
+import com.karumi.dexter.listener.PermissionRequest
+import com.karumi.dexter.listener.single.PermissionListener
+
+class GpsLocation(val context: Context) : LocationListener {
```

Comment:
Creo que deberíamos usar el contexto de aplicación (al igual que hace Antonio), lo que implicaría extraer el manejo de permisos (Dexter) a otra clase. 
Ver: 

- https://blog.mindorks.com/understanding-context-in-android-application-330913e32514

- Proyecto de ejemplo Architect Codders.

```suggestion
class GpsLocation(private val application: Application) : LocationListener {
```